xExtension-ColorfulList/static/script.js

```diff
@@ -0,0 +1,40 @@
+$(document).ready(function(){
+    function monitorEntry(monitorCallback) {
+        const targetNode = document.getElementById('stream');
+        const config = { attributes: false, childList: true, subtree: false};
+        const callback = function(mutationsList, observer) {
+            for(let mutation of mutationsList) {
+                if (mutation.type === 'childList') {
+                    monitorCallback(mutationsList);
+                }
+                    }
+        };
+        const observer = new MutationObserver(callback);
+        observer.observe(targetNode, config);
+        //observer.disconnect();
+    };
+    monitorEntry(colorize);
+});
+
+
+function colorize(entries){
+    let entry = document.querySelectorAll('.flux_header');
+    entry.forEach((e,i)=>{
+        let cl = stringToColour(e.querySelector('.website').textContent)+'12';
```

Comment:
Incidentally, looking at all this I'd be more inclined to think in this direction:

```js
let css = [...document.querySelectorAll('.item.feed')].map(el => {
 let id = el.id.slice(2)
 return `.flux_header[data-feed=${id}] {
  background-color: ${stringToColor(el.textContent)};
 }`;
}.join(''); // Joining all the strings prevents creating a bigger and bigger string along the way.
// append CSS to page
```

Though when I phrase it like that it might make more sense to do that in PHP instead. :-)

Reply:
PS Unlike the jQuery thing that's not something to change before merging, just some general thoughts.

Reply:
Thanks @Frenzie ! And I might be a bit concerned to touch php part as actually I am not really familiar with php 😢 Hence maybe it is 'safer' for me to make such patch-like js mocking if as you mentioned not kind of serious concern there?😆

Reply:
Regardless of JS or PHP, I think the approach I sketched is simpler because you calculate once without needing a MutationObserver or any of that. ;-)

But I just realized the security policy may annoyingly disallow what I was proposing.
```
let stylesheet = document.createElement('style');
stylesheet.innerText = css;
document.head.appendChild(stylesheet);
```

Reply:
> Regardless of JS or PHP, I think the approach I sketched is simpler because you calculate once without needing a MutationObserver or any of that. ;-)
> 
> But I just realized the security policy may annoyingly disallow what I was proposing.
> 
> ```js
> let stylesheet = document.createElement('style');
> stylesheet.innerText = css;
> document.head.appendChild(stylesheet);
> ```

OK, but in case following proposal above, my concern is, once we remove MutationObserver , sounds the pre-created css style will only cover the first page's list ,i.e. the style loaded later with different feed source will then not be covered by those pre(and one-shot) css styles? 

Or, in that sense, I need to generate those css styles based on fetch all feeds info in advance but not on the fly during the page load? Of course, it's doable, but, somehow it's not so 'lazy' and straightforward like now...Maybe I wrongly read your suggestion as amateur....thanks again for your time and patient on reviewing this :)

,

Reply:
`document.querySelectorAll('.item.feed')` is the menu, or in other words every single feed there is.

Reply:
Yes, you are right -> I wrongly thought it's grabbing from the post list panel! thanks!